src/rdsn/include/dsn/utility/metrics.h

```diff
@@ -168,6 +171,16 @@ class metric_entity : public ref_counter
 
     ~metric_entity();
 
+    // Close all "closeable" metrics owned by this entity.
+    //
+    // `async` is used to control if the close operations are asynchronous or not. It is set to
+    // true by default, which means all metrics owned by this entity will be closed asynchronously
+    // without waiting for the close operations to be finished.
+    //
+    // Otherwise, once `async` is set to false, close() will be blocked until the close operations
+    // are finished.
+    void close(bool async = true);
```

Comment:
use 'wait' maybe more accurate, and it's suggest to use enum class instead of boolean, true/false is meanless when we see the invoker.

Reply:
OK, we can use enum class instead.